source/cppfront.cpp

```diff
@@ -361,17 +370,17 @@ class positional_printer
     {
         source_has_cpp2 = has_cpp2;
         cpp2_filename = cpp2_filename_;
-        assert (!out.is_open() && !pcomments && "ICE: tried to call .open twice");
+        assert (!file_out.is_open() && !pcomments && "ICE: tried to call .open twice");
         cpp1_filename = cpp1_filename_;
-        out.open(cpp1_filename);
+        if (!cpp1_filename.empty()) file_out.open(cpp1_filename);
         pcomments = &comments;
     }
 
     auto is_open() -> bool {
-        if (out.is_open()) {
+        if (file_out.is_open() || cpp1_filename.empty()) {
             assert (pcomments && "ICE: if out.is_open, pcomments should also be set");
         }
-        return out.is_open();
+        return file_out.is_open() || cpp1_filename.empty();
     }
```

Comment:
I'd suggest moving the `cpp1_filename.empty()` tests to a `is_stdout_output()` utility function for clarity.